field_manager/field_manager.F90

```diff
@@ -5058,47 +3617,30 @@ function fm_copy_list(list_name, suffix, create ) &
 
 num_meth= 1
 list_name_new = trim(list_name)//trim(suffix)
-!
   recursive_t = .true.
-!
 !        Initialize the field manager if needed
-!
-if (.not. module_is_initialized) then  !{
+if (.not. module_is_initialized) then
   call initialize
-endif  !}
+endif
 
-if (list_name .eq. ' ') then  !{
-!
+if (list_name .eq. ' ') then
 !        If list is empty, then dump the current list
-!
   temp_list_p => current_list_p
   success = .true.
-else  !}{
-!
+else
 !        Get a pointer to the list
-!
   temp_list_p => find_list(list_name, current_list_p, .false.)
-  if (associated(temp_list_p)) then  !{
+  if (associated(temp_list_p)) then
     success = .true.
-  else  !}{
-!
+  else
 !        Error following the path
-!
-    if (verb .gt. verb_level_warn) then
-      write (out_unit,*) trim(warn_header), 'Could not follow path for ', trim(list_name)
-    endif
     success = .false.
-  endif  !}
-endif  !}
-
-!
+  endif
+endif
 !        Find the list
-!
-if (success) then  !{
-  method(:) = ' '
-  control(:) = ' '
```

Comment:
should these two strings be initialized in the if below?

Reply:
I'm not sure what to do here. In the part you referenced, method and control have a fixed dimension of MAX_FIELD_METHODS. However, in theory, we'd want these to be allocatable according to whatever the actual length of control and method are from the result of fm_find_methods. I think, although I'm not sure, that the output of the fm_find_methods does the allocation, so we don't have to worry about it going unallocated. it would then just be a question of if we think not having a "default" value for them would be dangerous. I can't add those lines in as is because they aren't allocated until the fm_find_methods routine is called. Any alternatives you can think of or should we just leave it?

Reply:
Yeah i see what you mean. I don't see an allocation anywhere but i think since it's intent(out) and written to in `find_methods` it works out. Maybe it should be allocated and initialized in find_methods before it's written to, but that function is recursive so that might be a bad idea and it seems to be working fine. I guess we should leave it as is.